From cf48b9e5e8248e338ba4d639643ca136ff6e68c4 Mon Sep 17 00:00:00 2001
From: Yu Liao <liaoyu15@huawei.com>
Date: Fri, 31 Dec 2021 15:59:19 +0800
Subject: [PATCH] kabi: reserve space for time and workqueue subsystem related
 structure
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: cf48b9e5e8248e338ba4d639643ca136ff6e68c4


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4GUAB
CVE: NA

-------------------------------

Reserve space for timer and workqueue subsystem.

Signed-off-by: Yu Liao <liaoyu15@huawei.com>
Reviewed-by: wangxiongfeng 00379786 <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/timer.h     |  6 ++++++
 include/linux/workqueue.h | 11 +++++++++++
 2 files changed, 17 insertions(+)

diff --git a/include/linux/timer.h b/include/linux/timer.h
index d10bc7e73b41..d2dc1cb8b9f8 100644
--- a/include/linux/timer.h
+++ b/include/linux/timer.h
@@ -2,6 +2,7 @@
 #ifndef _LINUX_TIMER_H
 #define _LINUX_TIMER_H
 
+#include <linux/kabi.h>
 #include <linux/list.h>
 #include <linux/ktime.h>
 #include <linux/stddef.h>
@@ -21,6 +22,11 @@ struct timer_list {
 #ifdef CONFIG_LOCKDEP
 	struct lockdep_map	lockdep_map;
 #endif
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 #ifdef CONFIG_LOCKDEP
diff --git a/include/linux/workqueue.h b/include/linux/workqueue.h
index a48e8ea06d74..4ecfecb1202f 100644
--- a/include/linux/workqueue.h
+++ b/include/linux/workqueue.h
@@ -6,6 +6,7 @@
 #ifndef _LINUX_WORKQUEUE_H
 #define _LINUX_WORKQUEUE_H
 
+#include <linux/kabi.h>
 #include <linux/timer.h>
 #include <linux/linkage.h>
 #include <linux/bitops.h>
@@ -106,6 +107,11 @@ struct work_struct {
 #ifdef CONFIG_LOCKDEP
 	struct lockdep_map lockdep_map;
 #endif
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 #define WORK_DATA_INIT()	ATOMIC_LONG_INIT((unsigned long)WORK_STRUCT_NO_POOL)
@@ -121,6 +127,11 @@ struct delayed_work {
 	int cpu;
 	/* delayed_work private data, only used in pciehp now */
 	unsigned long data;
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 struct rcu_work {
-- 
2.26.2

